### ISSUE TITLE
`jaxtyped(typechecked(..))` doesn't seem to work with generators

### ISSUE THREAD

**Issue by ar0ck** [patrick-kidger/jaxtyping#91]
Consider the following PyTorch repro:
```python
import jaxtyping
import torch
import typeguard
from typing import Iterator

@jaxtyping.jaxtyped
@typeguard.typechecked
def g(x: jaxtyping.Shaped[torch.Tensor, '*']) -> Iterator[jaxtyping.Shaped[torch.Tensor, '*']]:
    yield x

@jaxtyping.jaxtyped
def f():
    next(g(torch.zeros(1)))
    next(g(torch.zeros(2)))

f()
```

This yields `TypeError: type of value yielded from generator must be jaxtyping.Shaped[Tensor, '*']; got torch.Tensor instead`, due to [this condition](https://github.com/google/jaxtyping/blob/1b9c9fab5298a0462d92eba2986323001e6600b1/jaxtyping/_array_types.py#L257) failing with `torch.Size([1]) != torch.Size([2])`.

**Comment by patrick-kidger** [comment]
Hmm. It looks like the inner `jaxtyped` is simply applied to starting the generator (i.e. it's arguments). But as values aren't yielded until later, they're all done in the dynamic context of the outer `jaxtyped`.

I suppose we should check the output of the function, see if it's a generator, and wrap it if so.

I'd welcome a PR on this.

**Comment by patrick-kidger** [comment]
Fixed in version 0.2.26!
